src/HazelcastClient.ts

```diff
@@ -68,6 +71,22 @@ class HazelcastClient {
         return deferred.promise;
     }
 
+    getDistributedObjects(): Q.Promise<DistributedObject[]> {
+        var deferred = Q.defer<DistributedObject[]>();
+        var clientMessage = ClientGetDistributedObjectsCodec.encodeRequest();
+        var toObjectFunc = this.serializationService.toObject.bind(this);
+        var proxyManager = this.proxyManager;
+        this.invocationService.invokeOnRandomTarget(clientMessage).then(function(resp) {
+            var objectsInfoList = ClientGetDistributedObjectsCodec.decodeResponse(resp, toObjectFunc).response;
+            var proxies: DistributedObject[] = [];
+            for (var i = 0; i < objectsInfoList.size(); i++)  {
+                proxies.push(proxyManager.getOrCreateProxy(objectsInfoList.get(i)[1], objectsInfoList.get(i)[0]));
```

Comment:
getOrCreateProxy does not create a new proxy if it already exists.
All proxies may not be present in this client. Another client may have created a proxy at the server.